onnx/onnxifi_loader.h

```diff
@@ -0,0 +1,106 @@
+#define ONNXIFI_LOADER_FLAG_VERSION_1_0 0x01
+
+#ifndef ONNXIFI_HIDDEN
+#if defined(__ELF__)
+#define ONNXIFI_HIDDEN __attribute__((__visibility__("hidden")))
+#elif defined(__MACH__)
+#define ONNXIFI_HIDDEN __attribute__((__visibility__("hidden")))
+#else
+#define ONNXIFI_HIDDEN
+#endif
+#endif /* !defined(ONNXIFI_HIDDEN) */
+
+struct onnxifi_library {
+  /*
+   * Opaque handle for the loaded ONNXIFI library.
+   *
+   * Note: this is the value returned from LoadLibraryW (on Windows), or
+   * dlopen (on other operating systems and environments).
+   */
+  void* handle;
+  /*
+   * Options used for dynamic loading of the ONNXIFI library and its API
+   * functions. These are the options passed in flags parameter to onnxifi_load.
+   */
+  uint32_t flags;
+  union {
+    struct {
+      onnxGetBackendIDsFunction onnxGetBackendIDs;
+      onnxReleaseBackendIDFunction onnxReleaseBackendID;
+      onnxGetBackendInfoFunction onnxGetBackendInfo;
+      onnxGetBackendCompatibilityFunction onnxGetBackendCompatibility;
+      onnxInitBackendFunction onnxInitBackend;
+      onnxReleaseBackendFunction onnxReleaseBackend;
+      onnxInitEventFunction onnxInitEvent;
+      onnxSignalEventFunction onnxSignalEvent;
+      onnxWaitEventFunction onnxWaitEvent;
+      onnxReleaseEventFunction onnxReleaseEvent;
+      onnxInitGraphFunction onnxInitGraph;
+      onnxSetGraphIOFunction onnxSetGraphIO;
+      onnxRunGraphFunction onnxRunGraph;
+      onnxReleaseGraphFunction onnxReleaseGraph;
+    };
+    void* functions[ONNXIFI_LOADER_FUNCTION_COUNT];
+  };
+};
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Dynamically load the ONNXIFI library.
+ *
+ * @param[out] library - the structure representing the dynamic library and
+ *                       its API functions. On success, this structure will
+ *                       be initialized with valid pointers to implentation.
+ *                       On failure, this structure will be zero-initialized.
+ * @param flags - options for dynamic loading of the ONNXIFI library.
+ *                The ONNXIFI_LOADER_FLAG_VERSION_MASK part of the flags
+ *                specifies which ONNX API function should be loaded. The only
+ *                currently supported value is ONNXIFI_LOADER_FLAG_VERSION_1_0.
+ * @param path - optional path to the ONNXIFI library to load.
+ *               If this argument is null, the default path is used.
+ * @param[in] suffix - optional suffix for names of ONNX Backend API functions.
+ *                     E.g. if this argument is "GAMMA", the loader will search
+ *                     for "onnxGetNumBackendsGAMMA", "onnxGetBackendInfoGAMMA",
+ *                     etc. functions in the loaded library.
+ *                     If this argument is null, an empty suffix is used.
+ *
+ * @return Non-zero if the function succeeds, or zero if the function fails.
+ */
+ONNXIFI_HIDDEN int ONNXIFI_ABI onnxifi_load(
+  struct onnxifi_library* library,
```

Comment:
out parameter may be put at the end?

Reply:
Moved to the end of argument list. Attn @yinghai for change in the interface.